### ISSUE TITLE
tests fail to build

### ISSUE THREAD

**Issue by matthiaskrgr** [wesnoth/wesnoth#1900]
Hi, looks like one file (tests) still fails building after yesterdays fixes:
````
[ 75%/3/1/4,8.468] Building CXX object src/CMakeFiles/test.dir/tests/test_irdya_date.cpp.o
FAILED: src/CMakeFiles/test.dir/tests/test_irdya_date.cpp.o 
/home/matthias/LLVM/LLVM_dev/stage_2/build/bin/clang++  -DBOOST_TEST_DYN_LINK -DHAS_RELATIVE_LOCALEDIR=1 -DHAVE_CONFIG_H -DHAVE_FRIBIDI -DHAVE_LIBDBUS -DHAVE_LIBPNG -DLUA_USE_POSIX -D_X11 -isystem /usr/include/freetype2 -isystem /usr/include/libpng16 -isystem /usr/include/pango-1.0 -isystem /usr/include/harfbuzz -isystem /usr/include/cairo -isystem /usr/include/glib-2.0 -isystem /usr/lib64/glib-2.0/include -isystem /usr/include/pixman-1 -isystem /usr/include/libdrm -isystem /usr/include/dbus-1.0 -isystem /usr/lib64/dbus-1.0/include -isystem /usr/include/SDL2 -I../src -Isrc -isystem /usr/include/fribidi -std=c++11 -W -Wall -fsanitize=address,undefined -g3 -I/usr/include/boost -Og -fno-omit-frame-pointer -march=native -fsanitize-address-use-after-scope  -fno-optimize-sibling-calls -O2   -I/usr/include/SDL2 -D_REENTRANT -DWESNOTH_PATH=\"/usr/local/share/wesnoth\" -DLOCALEDIR=\"translations\" -DFIFODIR=\"/var/run/wesnothd\" -MD -MT src/CMakeFiles/test.dir/tests/test_irdya_date.cpp.o -MF src/CMakeFiles/test.dir/tests/test_irdya_date.cpp.o.d -o src/CMakeFiles/test.dir/tests/test_irdya_date.cpp.o -c ../src/tests/test_irdya_date.cpp
In file included from ../src/tests/test_irdya_date.cpp:17:
In file included from /usr/include/boost/test/unit_test.hpp:18:
In file included from /usr/include/boost/test/test_tools.hpp:46:
In file included from /usr/include/boost/test/tools/old/impl.hpp:24:
/usr/include/boost/test/tools/detail/print_helper.hpp:50:14: error: invalid operands to binary expression ('std::ostream' (aka 'basic_ostream<char>') and 'const irdya_date')
        ostr << t;
        ~~~~ ^  ~
/usr/include/boost/test/tools/detail/print_helper.hpp:172:5: note: in instantiation of member function 'boost::test_tools::tt_detail::print_log_value<irdya_date>::operator()' requested here
    print_log_value<T>()( ostr, ph.m_t );
    ^
/usr/include/boost/test/utils/lazy_ostream.hpp:66:29: note: in instantiation of function template specialization 'boost::test_tools::tt_detail::operator<<<irdya_date>' requested here
        return m_prev(ostr) << m_value;
                            ^
/usr/include/boost/test/tools/old/impl.hpp:92:50: note: in instantiation of member function 'boost::unit_test::lazy_ostream_impl<boost::unit_test::lazy_ostream, boost::test_tools::tt_detail::print_helper_t<irdya_date>, const boost::test_tools::tt_detail::print_helper_t<irdya_date> &>::operator()' requested here
BOOST_PP_REPEAT( BOOST_TEST_MAX_PREDICATE_ARITY, IMPL_FRWD, _ )
                                                 ^
/usr/include/boost/preprocessor/repetition/repeat.hpp:29:26: note: expanded from macro 'BOOST_PP_REPEAT'
# define BOOST_PP_REPEAT BOOST_PP_CAT(BOOST_PP_REPEAT_, BOOST_PP_AUTO_REC(BOOST_PP_REPEAT_P, 4))
                         ^
/usr/include/boost/preprocessor/cat.hpp:22:32: note: expanded from macro 'BOOST_PP_CAT'
#    define BOOST_PP_CAT(a, b) BOOST_PP_CAT_I(a, b)
                               ^
/usr/include/boost/preprocessor/cat.hpp:29:34: note: expanded from macro 'BOOST_PP_CAT_I'
#    define BOOST_PP_CAT_I(a, b) a ## b
                                 ^
<scratch space>:134:1: note: expanded from here
BOOST_PP_REPEAT_1
^
../src/tests/test_irdya_date.cpp:40:2: note: in instantiation of function template specialization 'boost::test_tools::tt_detail::check_frwd<boost::test_tools::tt_detail::equal_impl_frwd, irdya_date, irdya_date>' requested here
        BOOST_CHECK_EQUAL(first, second);
        ^
/usr/include/boost/test/tools/old/interface.hpp:152:45: note: expanded from macro 'BOOST_CHECK_EQUAL'
#define BOOST_CHECK_EQUAL( L, R )           BOOST_TEST_TOOL_IMPL( 0, \
                                            ^
/usr/include/boost/test/tools/old/interface.hpp:65:47: note: expanded from macro 'BOOST_TEST_TOOL_IMPL'
    BOOST_PP_IF( frwd_type, report_assertion, check_frwd ) (                    \
                                              ^
/usr/lib/gcc/x86_64-redhat-linux/6.4.1/../../../../include/c++/6.4.1/ostream:245:7: note: candidate function not viable: no known conversion from 'const irdya_date' to 'const void *' for 1st argument; take the address of the argument with &
      operator<<(const void* __p)
      ^
/usr/lib/gcc/x86_64-redhat-linux/6.4.1/../../../../include/c++/6.4.1/system_error:209:5: note: candidate function not viable: no known conversion from 'const irdya_date' to 'const std::error_code' for 2nd argument
    operator<<(basic_ostream<_CharT, _Traits>& __os, const error_code& __e)
    ^
/usr/lib/gcc/x86_64-redhat-linux/6.4.1/../../../../include/c++/6.4.1/ostream:108:7: note: candidate function not viable: no known conversion from 'const irdya_date' to 'std::basic_ostream<char, std::char_traits<char> >::__ostream_type &(*)(std::basic_ostream<char, std::char_traits<char> >::__ostream_type &)' (aka 'basic_ostream<char, std::char_traits<char> > &(*)(basic_ostream<char, std::char_traits<char> > &)') for 1st argument
      operator<<(__ostream_type& (*__pf)(__ostream_type&))
      ^
/usr/lib/gcc/x86_64-redhat-linux/6.4.1/../../../../include/c++/6.4.1/ostream:117:7: note: candidate function not viable: no known conversion from 'const irdya_date' to 'std::basic_ostream<char, std::char_traits<char> >::__ios_type &(*)(std::basic_ostream<char, std::char_traits<char> >::__ios_type &)' (aka 'basic_ios<char, std::char_traits<char> > &(*)(basic_ios<char, std::char_traits<char> > &)') for 1st argument
      operator<<(__ios_type& (*__pf)(__ios_type&))
      ^
/usr/lib/gcc/x86_64-redhat-linux/6.4.1/../../../../include/c++/6.4.1/ostream:127:7: note: candidate function not viable: no known conversion from 'const irdya_date' to 'std::ios_base &(*)(std::ios_base &)' for 1st argument
      operator<<(ios_base& (*__pf) (ios_base&))
      ^
/usr/lib/gcc/x86_64-redhat-linux/6.4.1/../../../../include/c++/6.4.1/ostream:166:7: note: candidate function not viable: no known conversion from 'const irdya_date' to 'long' for 1st argument
      operator<<(long __n)
      ^
/usr/lib/gcc/x86_64-redhat-linux/6.4.1/../../../../include/c++/6.4.1/ostream:170:7: note: candidate function not viable: no known conversion from 'const irdya_date' to 'unsigned long' for 1st argument
      operator<<(unsigned long __n)
      ^
/usr/lib/gcc/x86_64-redhat-linux/6.4.1/../../../../include/c++/6.4.1/ostream:174:7: note: candidate function not viable: no known conversion from 'const irdya_date' to 'bool' for 1st argument
      operator<<(bool __n)
      ^
/usr/lib/gcc/x86_64-redhat-linux/6.4.1/../../../../include/c++/6.4.1/ostream:178:7: note: candidate function not viable: no known conversion from 'const irdya_date' to 'short' for 1st argument
      operator<<(short __n);
      ^
/usr/lib/gcc/x86_64-redhat-linux/6.4.1/../../../../include/c++/6.4.1/ostream:181:7: note: candidate function not viable: no known conversion from 'const irdya_date' to 'unsigned short' for 1st argument
      operator<<(unsigned short __n)
      ^
/usr/lib/gcc/x86_64-redhat-linux/6.4.1/../../../../include/c++/6.4.1/ostream:189:7: note: candidate function not viable: no known conversion from 'const irdya_date' to 'int' for 1st argument
      operator<<(int __n);
      ^
/usr/lib/gcc/x86_64-redhat-linux/6.4.1/../../../../include/c++/6.4.1/ostream:192:7: note: candidate function not viable: no known conversion from 'const irdya_date' to 'unsigned int' for 1st argument
      operator<<(unsigned int __n)
      ^
/usr/lib/gcc/x86_64-redhat-linux/6.4.1/../../../../include/c++/6.4.1/ostream:201:7: note: candidate function not viable: no known conversion from 'const irdya_date' to 'long long' for 1st argument
      operator<<(long long __n)
      ^
/usr/lib/gcc/x86_64-redhat-linux/6.4.1/../../../../include/c++/6.4.1/ostream:205:7: note: candidate function not viable: no known conversion from 'const irdya_date' to 'unsigned long long' for 1st argument
      operator<<(unsigned long long __n)
      ^
/usr/lib/gcc/x86_64-redhat-linux/6.4.1/../../../../include/c++/6.4.1/ostream:220:7: note: candidate function not viable: no known conversion from 'const irdya_date' to 'double' for 1st argument
      operator<<(double __f)
      ^
/usr/lib/gcc/x86_64-redhat-linux/6.4.1/../../../../include/c++/6.4.1/ostream:224:7: note: candidate function not viable: no known conversion from 'const irdya_date' to 'float' for 1st argument
      operator<<(float __f)
      ^
/usr/lib/gcc/x86_64-redhat-linux/6.4.1/../../../../include/c++/6.4.1/ostream:232:7: note: candidate function not viable: no known conversion from 'const irdya_date' to 'long double' for 1st argument
      operator<<(long double __f)
      ^
/usr/lib/gcc/x86_64-redhat-linux/6.4.1/../../../../include/c++/6.4.1/ostream:270:7: note: candidate function not viable: no known conversion from 'const irdya_date' to 'std::basic_ostream<char, std::char_traits<char> >::__streambuf_type *' (aka 'basic_streambuf<char, std::char_traits<char> > *') for 1st argument
      operator<<(__streambuf_type* __sb);
      ^
/usr/lib/gcc/x86_64-redhat-linux/6.4.1/../../../../include/c++/6.4.1/ostream:502:5: note: candidate function not viable: no known conversion from 'const irdya_date' to 'char' for 2nd argument
    operator<<(basic_ostream<_CharT, _Traits>& __out, char __c)
    ^
/usr/lib/gcc/x86_64-redhat-linux/6.4.1/../../../../include/c++/6.4.1/ostream:508:5: note: candidate function not viable: no known conversion from 'const irdya_date' to 'char' for 2nd argument
    operator<<(basic_ostream<char, _Traits>& __out, char __c)
    ^
/usr/lib/gcc/x86_64-redhat-linux/6.4.1/../../../../include/c++/6.4.1/ostream:514:5: note: candidate function not viable: no known conversion from 'const irdya_date' to 'signed char' for 2nd argument
    operator<<(basic_ostream<char, _Traits>& __out, signed char __c)
    ^
/usr/lib/gcc/x86_64-redhat-linux/6.4.1/../../../../include/c++/6.4.1/ostream:519:5: note: candidate function not viable: no known conversion from 'const irdya_date' to 'unsigned char' for 2nd argument
    operator<<(basic_ostream<char, _Traits>& __out, unsigned char __c)
    ^
/usr/lib/gcc/x86_64-redhat-linux/6.4.1/../../../../include/c++/6.4.1/ostream:556:5: note: candidate function not viable: no known conversion from 'const irdya_date' to 'const char *' for 2nd argument
    operator<<(basic_ostream<char, _Traits>& __out, const char* __s)
    ^
/usr/lib/gcc/x86_64-redhat-linux/6.4.1/../../../../include/c++/6.4.1/ostream:569:5: note: candidate function not viable: no known conversion from 'const irdya_date' to 'const signed char *' for 2nd argument
    operator<<(basic_ostream<char, _Traits>& __out, const signed char* __s)
    ^
/usr/lib/gcc/x86_64-redhat-linux/6.4.1/../../../../include/c++/6.4.1/ostream:574:5: note: candidate function not viable: no known conversion from 'const irdya_date' to 'const unsigned char *' for 2nd argument
    operator<<(basic_ostream<char, _Traits>& __out, const unsigned char* __s)
    ^
/usr/lib/gcc/x86_64-redhat-linux/6.4.1/../../../../include/c++/6.4.1/ostream:628:5: note: candidate function [with _CharT = char, _Traits = std::char_traits<char>, _Tp = irdya_date] not viable: no known conversion from 'std::ostream' (aka 'basic_ostream<char>') to 'basic_ostream<char, std::char_traits<char> > &&' for 1st argument
    operator<<(basic_ostream<_CharT, _Traits>&& __os, const _Tp& __x)
    ^
/usr/lib/gcc/x86_64-redhat-linux/6.4.1/../../../../include/c++/6.4.1/bits/ostream.tcc:321:5: note: candidate function not viable: no known conversion from 'const irdya_date' to 'const char *' for 2nd argument
    operator<<(basic_ostream<_CharT, _Traits>& __out, const char* __s)
    ^
/usr/lib/gcc/x86_64-redhat-linux/6.4.1/../../../../include/c++/6.4.1/ostream:497:5: note: candidate template ignored: deduced conflicting types for parameter '_CharT' ('char' vs. 'irdya_date')
    operator<<(basic_ostream<_CharT, _Traits>& __out, _CharT __c)
    ^
/usr/lib/gcc/x86_64-redhat-linux/6.4.1/../../../../include/c++/6.4.1/bits/basic_string.h:5340:5: note: candidate template ignored: could not match 'basic_string<type-parameter-0-0, type-parameter-0-1, type-parameter-0-2>' against 'const irdya_date'
    operator<<(basic_ostream<_CharT, _Traits>& __os,
    ^
/usr/lib/gcc/x86_64-redhat-linux/6.4.1/../../../../include/c++/6.4.1/ostream:539:5: note: candidate template ignored: could not match 'const _CharT *' against 'irdya_date'
    operator<<(basic_ostream<_CharT, _Traits>& __out, const _CharT* __s)
    ^
/usr/include/boost/test/utils/wrap_stringstream.hpp:64:1: note: candidate template ignored: could not match 'basic_wrap_stringstream' against 'basic_ostream'
operator<<( basic_wrap_stringstream<CharT>& targ, T const& t )
^
/usr/include/boost/test/utils/wrap_stringstream.hpp:109:1: note: candidate template ignored: could not match 'basic_wrap_stringstream' against 'basic_ostream'
operator<<( basic_wrap_stringstream<CharT>& targ, basic_wrap_stringstream<CharT>& src )
^
/usr/include/boost/test/utils/wrap_stringstream.hpp:121:1: note: candidate template ignored: could not match 'basic_wrap_stringstream' against 'basic_ostream'
operator<<( basic_wrap_stringstream<CharT>& targ, std::ios_base& (BOOST_TEST_CALL_DECL *man)(std::ios_base&) )
^
/usr/include/boost/test/utils/wrap_stringstream.hpp:131:1: note: candidate template ignored: could not match 'basic_wrap_stringstream' against 'basic_ostream'
operator<<( basic_wrap_stringstream<CharT>& targ, std::basic_ostream<Elem,Tr>& (BOOST_TEST_CALL_DECL *man)(std::basic_ostream<Elem, Tr>&) )
^
/usr/include/boost/test/utils/wrap_stringstream.hpp:141:1: note: candidate template ignored: could not match 'basic_wrap_stringstream' against 'basic_ostream'
operator<<( basic_wrap_stringstream<CharT>& targ, std::basic_ios<Elem, Tr>& (BOOST_TEST_CALL_DECL *man)(std::basic_ios<Elem, Tr>&) )
^
/usr/lib/gcc/x86_64-redhat-linux/6.4.1/../../../../include/c++/6.4.1/bits/shared_ptr.h:66:5: note: candidate template ignored: could not match '__shared_ptr<type-parameter-0-2, _Lp>' against 'const irdya_date'
    operator<<(std::basic_ostream<_Ch, _Tr>& __os,
    ^
/usr/include/boost/smart_ptr/shared_ptr.hpp:897:64: note: candidate template ignored: could not match 'shared_ptr<type-parameter-0-2>' against 'const irdya_date'
template<class E, class T, class Y> std::basic_ostream<E, T> & operator<< (std::basic_ostream<E, T> & os, shared_ptr<Y> const & p)
                                                               ^
1 error generated.
ninja: build stopped: cannot make progress due to previous errors.

````

**Comment by gfgtdf** [comment]
@CelticMinstrel  mostlikeley like irdya_date lacks a operator<<(stream,irdya_date )

**Comment by Vultraz** [comment]
Was fixed with 78f0ef8.

**Comment by CelticMinstrel** [comment]
Thanks for the fix. I was away and couldn't address it, sorry.
